### PR TITLE
Debug acc for qwen3 moe

### PR DESCRIPTION
**Pull request by jayhenry**

### CODE REVIEW

File: xtuner/v1/engine/train_engine.py [hunk 1]
```diff
@@ -18,10 +17,9 @@
     set_optimizer_state_dict,
 )
 from torch.distributed.device_mesh import DeviceMesh
-from torch.distributed.tensor.placement_types import Placement
+from torch.nn.utils.clip_grad import _no_grad
```
Comment (HAOCHENYE) [inline xtuner/v1/engine/train_engine.py]: Why use `_no_grad` rather than `torch.no_grad` here?

File: xtuner/v1/engine/train_engine.py [hunk 2]
```diff
@@ -230,7 +230,10 @@ def train_step(self, data_batches: list[ModelItem]):
             self._count += 1
 
         train_engine_extra_info = ModelForwardExtraLogInfo()
+        micro_batch_iter = 0
         for i in range(0, len(data_batches), intra_layer_micro_batch):
+            ProberList.set_micro_batch_iter(micro_batch_iter)
+            micro_batch_iter += 1
```
Comment (HAOCHENYE) [inline xtuner/v1/engine/train_engine.py]:
```suggestion
            ProberList.set_micro_batch_iter(i)
```

File: xtuner/v1/loss/ce_loss.py [hunk 3]
```diff
@@ -11,6 +11,7 @@
 
 from xtuner.v1.loss import BaseLossConfig, BaseLossContext, BaseLossKwargs
 
+# from xtuner.v1.profiler.prober import ProberList
```
Comment (HAOCHENYE) [inline xtuner/v1/loss/ce_loss.py]: Clean

File: xtuner/v1/model/dense/qwen3.py
Comment (HAOCHENYE) [inline xtuner/v1/model/dense/qwen3.py]: Unnecessary declaration.

File: xtuner/v1/model/moe/qwen3.py [hunk 4]
```diff
@@ -123,7 +123,10 @@ def hf_config(self) -> HFQwen3MoeConfig:
 class Qwen3MoE30BA3Config(Qwen3MoEConfig):
     vocab_size: int = 151936
     max_position_embeddings: int = 40960
-    pad_token_id: int = 151643
+    # Qwen3 Model(dense and moe)'s pad_token_id is not set, so we need to set it to None.
```
Comment (HAOCHENYE) [inline xtuner/v1/model/moe/qwen3.py]: Only declaration in `Qwen3MoEConfig`

File: xtuner/v1/module/decoder_layer/moe_decoder_layer.py [hunk 5]
```diff
@@ -315,6 +314,9 @@ def _forward(
         origin_shape = hidden_states.shape
 
         # reshape hidden_states to (batch_size * seq_len, hidden_size)
+        # ProberList.before_dispatch(
```
Comment (HAOCHENYE) [inline xtuner/v1/module/decoder_layer/moe_decoder_layer.py]: Clean?
